test/units/testsuite-58.sh

```diff
@@ -84,6 +84,38 @@ cmp /var/tmp/testsuite-58.img /var/tmp/testsuite-58.2.img
 rm /var/tmp/testsuite-58.img /var/tmp/testsuite-58.2.img /tmp/testsuite-58.dump
 rm -r /tmp/testsuite-58-defs/
 
+# Third part: operate on an an image with unaligned partition, to see if that works.
+
+rm -f /var/tmp/testsuite-58.3.img /tmp/testsuite-58-3.dump
+mkdir -p /tmp/testsuite-58.3-defs/
+
+cat >/tmp/testsuite-58.3-defs/root.conf <<EOF
+[Partition]
+Type=root
+EOF
+
+truncate -s 10g /var/tmp/testsuite-58.3.img
```

Comment:
10g might be a bit harsh on some of the overcrowded testbeds we got - can the same functionality be tested with something smaller?

Reply:
it's a sparse file, it's free. (all linux fs know sparse files. i mean, the only common fs we deal with in linux that doesn't do sparse is FAT, but /var/tmp/ is not going to be that)

My goal here was to reuse the precise partition definition from the original bug report, which had such a large disk...

Consider it an extension to the test case: that the assumption that disk images that are sparse are really free holds, and that they are not made unsparse by systemd-repart. ;-)

Reply:
(by sparse file i mean: it's entirely empty except the blocks needed for the part table. i.e. a few KB at the beginning of the file and at the end, and everything in between is just one gigantic hole)

Reply:
ok, sounds good